content/zh/docs/topics/library_charts.md

```diff
@@ -0,0 +1,338 @@
+Update Complete. ⎈Happy Helming!⎈
+Saving 1 charts
+Deleting outdated charts
+```
+
+现在我们准备好部署chart了。安装之前，需要先检测渲染过的模板。
+
+```console
+$ helm install mydemo mychart/ --debug --dry-run
+install.go:159: [debug] Original chart version: ""
+install.go:176: [debug] CHART PATH: /root/test/helm-charts/mychart
+
+NAME: mydemo
+LAST DEPLOYED: Tue Mar  3 17:48:47 2020
+NAMESPACE: default
+STATUS: pending-install
+REVISION: 1
+TEST SUITE: None
+USER-SUPPLIED VALUES:
+{}
+
+COMPUTED VALUES:
+affinity: {}
+fullnameOverride: ""
+image:
+  pullPolicy: IfNotPresent
+  repository: nginx
+imagePullSecrets: []
+ingress:
+  annotations: {}
+  enabled: false
+  hosts:
+  - host: chart-example.local
+    paths: []
+  tls: []
+mylibchart:
+  global: {}
+nameOverride: ""
+nodeSelector: {}
+podSecurityContext: {}
+replicaCount: 1
+resources: {}
+securityContext: {}
+service:
+  port: 80
+  type: ClusterIP
+serviceAccount:
+  annotations: {}
+  create: true
+  name: null
+tolerations: []
+
+HOOKS:
+MANIFEST:
+---
+# Source: mychart/templates/configmap.yaml
+apiVersion: v1
+data:
+  myvalue: Hello World
+kind: ConfigMap
+metadata:
+  labels:
+    app: mychart
+    chart: mychart-0.1.0
+    release: mydemo
+  name: mychart-mydemo
+```
+
+这个看起来像是我们需要的用`myvalue: Hello World`覆盖的配置映射。现在安装：
+
+```console
+$ helm install mydemo mychart/
+NAME: mydemo
+LAST DEPLOYED: Tue Mar  3 17:52:40 2020
+NAMESPACE: default
+STATUS: deployed
+REVISION: 1
+TEST SUITE: None
+```
+
+我们可以检索这个版本并看到实际的版本已经加载。
+
+```console
+$ helm get manifest mydemo
+---
+# Source: mychart/templates/configmap.yaml
+apiVersion: v1
+data:
+  myvalue: Hello World
+kind: ConfigMap
+metadata:
+  labels:
+    app: mychart
+    chart: mychart-0.1.0
+    release: mydemo
+  name: mychart-mydemo
+  ```
+
+## The Common Helm Helper Chart
+
+满目都是标题但是这个[chart](https://github.com/helm/charts/tree/master/incubator/common)是公共chart的初始模式。
+它提供的应用程序反映了Kubernetes chart开发的最佳实践。最棒的是你开发chart时可以立即使用易用的共享代码。
+
+这里有一种快速使用它的方法。更多细节请查看[README](https://github.com/helm/charts/blob/master/incubator/common/README.md)。
+
+再创建一个脚手架：
+
+```console
+$ helm create demo
+Creating demo
+```
+
+使用辅助chart中的公共代码。首先编辑负载文件`demo/templates/deployment.yaml`如下：
+
+```yaml
+{{- template "common.deployment" (list . "demo.deployment") -}}
+{{- define "demo.deployment" -}}
+## Define overrides for your Deployment resource here, e.g.
+spec:
+  replicas: {{ .Values.replicaCount }}
+{{- end -}}
+```
+
+现在这个service文件`demo/templates/service.yaml`变成了下面这样：
+
+```yaml
+{{- template "common.service" (list . "demo.service") -}}
+{{- define "demo.service" -}}
+## Define overrides for your Service resource here, e.g.
+# metadata:
+#   labels:
+#     custom: label
+# spec:
+#   ports:
+#   - port: 8080
+{{- end -}}
+```
+
+这些模板显示了如何从辅助模板中继承公共代码来将你的代码简化到资源的配置或自定义。
+
+为了能使用公共代码，我们需要添加一个`common`依赖。在`demo/Chart.yaml`文件最后啊添加以下内容： 
+
+```yaml
+dependencies:
+- name: common
+  version: "^0.0.5"
+  repository: "https://kubernetes-charts-incubator.storage.googleapis.com/"
```

Comment:
Please update the repository URLs, as seen in https://github.com/helm/helm-www/pull/897.